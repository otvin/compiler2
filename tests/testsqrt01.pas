program testsqrt01(output);
{tests square root and the error around trying to take square root of a negative number}

var q:real;

function half(r:real):real;
begin
	half := r/2;
end;

begin
	writeln(half(42.0));
	writeln(sqrt(42.0));
	q := 74.65;
	if q < 0 then
		writeln('oops!')
	else
		begin
			q := sqrt(q);
			writeln(q)
		end;
	writeln(sqrt(100));
	writeln(sqrt(-5.7));
end.
