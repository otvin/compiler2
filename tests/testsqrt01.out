21.000000000000
6.480740698408
8.640023148117
10.000000000000
Error D.34: Cannot take sqrt() of negative number